command/agent/job_endpoint_test.go

```diff
@@ -1981,29 +1983,150 @@ func TestHTTP_JobDispatch(t *testing.T) {
 
 		// Make the HTTP request
 		req2, err := http.NewRequest(http.MethodPut, "/v1/job/"+job.ID+"/dispatch", buf)
+		must.NoError(t, err)
 		if err != nil {
-			t.Fatalf("err: %v", err)
+			t.Logf("err: %v", err)
 		}
 		respW.Flush()
 
 		// Make the request
 		obj, err := s.Server.JobSpecificRequest(respW, req2)
 		if err != nil {
-			t.Fatalf("err: %v", err)
+			t.Logf("err: %v", err)
 		}
 
 		// Check the response
 		dispatch := obj.(structs.JobDispatchResponse)
 		if dispatch.EvalID == "" {
-			t.Fatalf("bad: %v", dispatch)
+			t.Logf("bad: %v", dispatch)
 		}
-
+		must.NotEq(t, dispatch.EvalID, "")
 		if dispatch.DispatchedJobID == "" {
-			t.Fatalf("bad: %v", dispatch)
+			t.Logf("bad: %v", dispatch)
 		}
+		must.NotEq(t, dispatch.DispatchedJobID, "")
 	})
 }
 
+func TestHTTP_JobDispatchPriority(t *testing.T) {
+	ci.Parallel(t)
+	testCases := []struct {
+		name        string
+		priority    int
+		expectedErr bool
+		setPriority bool
+	}{{
+		name: "no priority",
+	},
+		{name: "set invalid priority",
+			priority:    103,
+			expectedErr: true,
+		},
+		{name: "set valid priority",
+			priority:    100,
+			setPriority: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			httpTest(t, nil, func(s *TestAgent) {
+				defaultPriority := 80
+				// Create the parameterized job
+				job := mock.BatchJob()
+				job.ParameterizedJob = &structs.ParameterizedJobConfig{}
+
+				job.Priority = defaultPriority
+				if tc.priority != 0 {
+					job.Priority = tc.priority
+				}
+				args := structs.JobRegisterRequest{
+					Job: job,
+					WriteRequest: structs.WriteRequest{
+						Region:    "global",
+						Namespace: structs.DefaultNamespace,
+					},
+				}
+				var resp structs.JobRegisterResponse
+				err := s.Agent.RPC("Job.Register", &args, &resp)
+				if !tc.expectedErr {
+					must.NoError(t, err)
+				} else {
+					must.Error(t, err)
+					must.StrContains(t, err.Error(), "job priority must be between")
+					return
+				}
+
+				// Make the request
+				respW := httptest.NewRecorder()
+				args2 := structs.JobDispatchRequest{
+					WriteRequest: structs.WriteRequest{
+						Region:           "global",
+						Namespace:        structs.DefaultNamespace,
+						IdempotencyToken: "foo",
+					},
+				}
+				buf := encodeReq(args2)
+
+				// Make the HTTP request
+				req2, err := http.NewRequest(http.MethodPut, "/v1/job/"+job.ID+"/dispatch", buf)
+				if err != nil {
+					t.Logf("err: %v", err)
+				}
+				must.NoError(t, err)
+				respW.Flush()
+
+				// Make the request
+				obj, err := s.Server.JobSpecificRequest(respW, req2)
+				if err != nil {
+					t.Logf("err: %v", err)
+				}
+				must.NoError(t, err)
```

Comment:
Like with the test above, there's a handful of cases here where we're logging instead of relying on the `must` assertion to report the results. We can remove all those.

Looks like we've got a bunch of cases of this in the `command/job_dispatch_test.go` file as well.